### DISCUSSION TITLE
Get Color, under the Cursor's character item ?

### DISCUSSION THREAD

**Discussion by paule32** [magiblot/tvision#151] [Q&A]
Hello,
like the Subject said: how can I get the Color Attributes under the Cursor - in a TEdit, TFileEdit derivated class ?

**Comment by paule32** [comment]
I found a partial Solution - So, implement a **std::vector< TColorAttr > shadowCopy** vector, that holds the Color Attributes of the Cells.

**Comment by magiblot** [comment]
`TEditor` is simple: it only has two color palette entries ([normal and selected text](https://github.com/magiblot/tvision/discussions/153#discussioncomment-8374651)). So, in the original `TEditor` it's as simple as checking whether the cursor is inside the current text selection or not.

However, since you are attempting to do syntax highlighting with several more colors, the solution is not so simple. The view's rendered contents are usually stored in the `buffer` member variable, but it is not reliable to try getting information from there because it depends on the `ofBuffered` flag in `options` being set and on whether `drawView()` has been already called or not (because it could otherwise contain outdated information).

So, a possible reasonable solution for a more complex use case like yours is what you already did: to have a vector or some other data structure storing the styling information of each cell in the text.
